Dockerfile

```diff
@@ -1,15 +1,17 @@
 # sha256:c2fda87e52bdedc1e3424e88187e24722793b4dc218e2be91d4501295387a02e
 FROM ghcr.io/emqx/docker-aws-cdk/aws-cdk:master
 
-RUN python3 -m venv /venv
-COPY requirements.txt /setup/requirements.txt
-COPY setup.py /setup/setup.py
+RUN curl -sSL https://install.python-poetry.org | POETRY_VERSION=1.1.14 python3 -
+ENV PATH=/root/.local/bin:$PATH
+WORKDIR /setup
+COPY pyproject.toml /setup/pyproject.toml
+COPY poetry.lock /setup/poetry.lock
+# install only cdk-emqx-cluster's deps for better docker caching
+RUN poetry install --no-root
 COPY README.md /setup/
 RUN mkdir -p /setup/cdk_emqx_cluster
-RUN cd /setup/ \
```

Comment:
I am not a Python pro but IMO with poetry the flow is quite the same as with `requirements.txt`
We
* Make venv
* Activate venv
* Install poetry inside venv with pip (`pip install poetry`)
* Having poetry and poetry.lock, now (inside venv) we run poetry install

Reply:
I'll try that alternative, but it seemed to me during the tests that it installed things only in its own virtual env, not the current one or globally.  And that needs to be activated with `poetry shell`.

But I'll try that as well to confirm.

Reply:
Some pythonistas use poetrty to manage venvs, but IMO this brings unnecessary complexity 😅

Reply:
Indeed, that worked.  I must've made some mistake while testing :see_no_evil: 

https://github.com/emqx/cdk-emqx-cluster/pull/96/commits/27b73f5d9b19c5a542beccca41eb171b7acf0911